akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -148,6 +148,16 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
       event.getClass.getName, seqNr, persistenceId, cause.getMessage)
   }
 
+  /**
+    * Called when the size of the internal stash is exceeds the capacity of the Stash
+    * 
+    * @param message the message which failed to enqueue
+    */
+  protected def onStashOverflow(message: Any): Unit = {
```

Comment:
two ways:
1. write some warning comment of above method to prompt users. constraint weakly but more flexible, imagine(may be not appropriately):  some users need log this event in custom format.
2. or more strictly, predefine sealed ```StashOverflowStrategy``` trait as @hepin1989 said. and  then provide ```DiscardToDeadLettersStrategy```,  ```ThrowOverflowExceptionStrategy```, ```ReplyRejectMessageStrategy```

How are u feeling?